snuba/datasets/events.py

```diff
@@ -29,6 +30,9 @@ class EventsDataset(TimeSeriesDataset):
 
     def __init__(self) -> None:
 
+        storage = get_storage("events")
+        assert isinstance(storage, WritableTableStorage)
```

Comment:
This sucks but the plan builder needs a ReadableStorage and now mypy only knows that it has some kind of Storage.

Reply:
you can have two factory methods, one for writes and one for generic storages.

Reply:
yeah, thinking over this, I think having different factory methods with specific types is going to be a good solution. It does not cost much and it will allow us to catch issues during refactoring through type checking (which means automatically) instead than at runtime. Imagine if we turned a storage from a writable to a readable one. mypy will immediately tell us what we need to change instead of relying on a runtime error.

Reply:
+1 on explicitly separating writable and readable (technically read/write) storages into different registries.

Reply:
Sounds good